### ISSUE TITLE
[C+ Checklist Needs Completion] [$500] Profile - Incorrect city name in automatic timezone selection

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#30468]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.91-6
**Reproducible in staging?:** Y
**Reproducible in production?:**Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
Pre-condition: User logged in and located in Ukraine
1. Open test website https://staging.new.expensify.com/
2. Open Settings
3. Click on your avatar
4. Click on the Timezone
5. Turn off "Automatically determine your location."
6. Click on the timezone selection
7. Type city exactly as in the automatical result "Kiev"

## Expected Result:
Name of the city in the manual and automatic time zone selection will be identical

## Actual Result:
Name of the city in the automatic timezone selection is Kiev, but if you try to find this timezone manually you can't find anything. Because timezone name in manual selection is Kyiv

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] Windows: Chrome 
- [x] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>Windows: Chrome </summary>

 

https://github.com/Expensify/App/assets/78819774/36362b81-2b78-4643-b883-0279f2516310

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01bb82fb7cffd6353f</li>
        <li>Upwork Job ID: 1717648944610619392</li>
        <li>Last Price Increase: 2023-11-16</li>
<li>Automatic offers: </li>
<ul>
<li>suneox | Contributor | 28061700</li>
</ul></li>
    </ul>
</details>

**Comment by iiredalert** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
We are unable to find Kiev as a timezone when selecting the time zone manually

### What is the root cause of that problem?

Kiev is not defined in TIMEZONES.js, only Kyiv is defined.

### What changes do you think we should make in order to solve the problem?

Add to TIMEZONES.js
```
    'Europe/Kiev',
```

### What alternative solutions did you explore? (Optional)

**Comment by girafferz** [comment]
# Proposal
## Please re-state the problem that we are trying to solve in this issue.
The relevant timezone is not included in the configured data.

## What is the root cause of that problem?
The timezone list is not being updated.

## What changes do you think we should make in order to solve the problem?
In JavaScript, you can use the moment-timezone library to get a list of time zones.

To automatically update the timezones.js file with the list of time zones obtained from moment.tz.names(), you have several options. One way is to create a script that dynamically generates the TIMEZONES.js file

When you run this script, it will generate (or update) the TIMEZONES.js file at the specified path. If the moment-timezone library is updated with new time zones, you can re-run this script to automatically update TIMEZONES.js.

Instead of running this script manually, you could also use some automation tool (e.g., a cron job or a CI/CD pipeline) to run it periodically.

example
```
const fs = require('fs');
const moment = require('moment-timezone');

const timezones = moment.tz.names();
let output = "export default [\n";

for (const tz of timezones) {
    output += `    '${tz}',\n`;
}

output += "];\n";

fs.writeFile('path/to/timezones.js', output, (err) => {
    if (err) {
        console.error('Error writing file:', err);
    } else {
        console.log('timezones.js has been updated');
    }
});

```

**Comment by girafferz** [comment]
Contributor details
Your Expensify account email: noiman2004@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~012a6fdebfafd446f0

**Comment by greg-schroeder** [comment]
Awaiting proposal review

**Comment by sobitneupane** [comment]
Thanks for the proposal everyone. At first, I would suggest everyone to go through the discussion in https://github.com/Expensify/App/pull/24446 PR. https://github.com/Expensify/App/pull/24446 PR was the one adding `TIMEZONES.js`. It has the explanation on why some timezones were not added.

**Comment by greg-schroeder** [comment]
Looks like we're waiting for some proposals to be updated after factoring in the comment above, I think

**Comment by greg-schroeder** [comment]
Awaiting updated proposals

**Comment by greg-schroeder** [comment]
Same as above

**Comment by sobitneupane** [comment]
@iiredalert @ZhenjaHorbach @suneox @girafferz

Did any of you get chance to review discussion in https://github.com/Expensify/App/pull/24446? Would you like to update your proposal?

https://github.com/Expensify/App/issues/30468#issuecomment-1794175163

**Comment by greg-schroeder** [comment]
bump @sobitneupane

**Comment by sobitneupane** [comment]
> We should get data from a [timezone database](https://en.wikipedia.org/wiki/Tz_database) and I have parse [List of tz database time zones](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones) upload file bellow we can replace for [TIMEZONES.js](https://github.com/Expensify/App/files/12741922/TIMEZONES.txt)
 
 [This](https://github.com/Expensify/App/pull/24446#issuecomment-1684096539) is the reason why we are not using some timezones. @suneox It looks like there are some timezones in https://en.wikipedia.org/wiki/List_of_tz_database_time_zones which IANA is not supporting.
 
 Specifically speaking, will the list contain timezones like `America/Buenos_Aires` and `Europe/Kiev`? Won't it cause a crash in IOS?

**Comment by sobitneupane** [comment]
@suneox Thanks for the update.

So, what we are doing is instead of showing `Europe/Kiev` when automatically detecting the timezone, we will be showing `Europe/Kyiv`, right?

The other way will be adding timezones like `Europe/Kiev` in Timezone Select page. Overall the proposal looks good, but I am not sure whether it is worth fixing.

The changes to introduce timezone.ts was introduced by @waterim in this [PR](https://github.com/Expensify/App/pull/24446#issuecomment-1684096539). So, I would like to know this view about this issue and this [proposal](https://github.com/Expensify/App/issues/30468#issuecomment-1782518130) when he has the opportunity.

**Comment by greg-schroeder** [comment]
discussing ongoing

**Comment by sobitneupane** [comment]
Thanks @suneox.

> when the user sets automatically detecting the timezone from the browser or another device and then accesses some feature related to tz it will crash on iOS

Have you tested this @suneox ? As per my testing app won't crash on such cases.

**Comment by waterim** [comment]
I think proposal on which I was mentioned make sense, if for some reason autoselection gets an outdated timezone - we need ti update it with supported one

**Comment by sobitneupane** [comment]
> I think proposal on which I was mentioned make sense, if for some reason autoselection gets an outdated timezone - we need ti update it with supported one

Thanks @waterim

**Comment by sobitneupane** [comment]
Thanks @suneox. If this bug might result in a crash for some user, we should get it solved asap. 

[Proposal](https://github.com/Expensify/App/issues/30468#issuecomment-1830361085) from @suneox looks good.

🎀 👀 🎀 C+ reviewed

**Comment by amyevans** [comment]
👍 to the proposal

**Comment by greg-schroeder** [comment]
Looks like @sobitneupane is still in the review phase!

**Comment by greg-schroeder** [comment]
Reviews continue on linked PR

**Comment by greg-schroeder** [comment]
Issue reported by: Applause
Contributor: @suneox ($500)
C+: @sobitneupane ($500) 

Paid @suneox, and you can make a manual request Sobit. All that's left is to complete the checklist above!

**Comment by greg-schroeder** [comment]
Relax Melvin

**Comment by greg-schroeder** [comment]
Bump again so we can get this one closed out!

**Comment by sobitneupane** [comment]
Sorry for the delay @greg-schroeder I was mostly unavailable last week. I will try to get it done this week (probably tomorrow)

**Comment by greg-schroeder** [comment]
No problem! Thanks Sobit!

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [ ]  [@sobitneupane] The PR that introduced the bug has been identified. Link to the PR:

https://github.com/Expensify/App/pull/24446
> * [ ]  [@sobitneupane] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

This issue is not solely caused by the [linked PR](https://github.com/Expensify/App/issues/30468#issuecomment-1833454429). Auto-selecting timezone is returning outdated timezone. Discussion: https://github.com/Expensify/App/issues/30468#issuecomment-1833454429 and https://github.com/Expensify/App/issues/30468#issuecomment-1850271581
> * [ ]  [@sobitneupane] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

It is an edge case that was missed while testing the PR.
> * [ ]  [@sobitneupane] Determine if we should create a regression test for this bug.

Nope. It was reported by Applause - Internal Team. So, I believe the regression test for this issue is already covered.

**Comment by greg-schroeder** [comment]
Thanks! closing

**Comment by JmillsExpensify** [comment]
$500 approved for @sobitneupane based on [this summary](https://github.com/Expensify/App/issues/30468#issuecomment-1896685873).
